samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/MainApplication.kt

```diff
@@ -0,0 +1,47 @@
+package io.customer.android.sample.kotlin_compose
+
+import android.app.Application
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import io.customer.android.sample.kotlin_compose.util.PreferencesKeys.API_KEY
+import io.customer.android.sample.kotlin_compose.util.PreferencesKeys.SITE_ID
+import io.customer.android.sample.kotlin_compose.util.PreferencesKeys.TRACK_API_URL_KEY
+import io.customer.messaginginapp.ModuleMessagingInApp
+import io.customer.messagingpush.ModuleMessagingPushFCM
+import io.customer.sdk.CustomerIO
+import io.customer.sdk.util.CioLogLevel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+
+class MainApplication : Application() {
+
+    @Inject
+    lateinit var dataStore: DataStore<Preferences>
+    override fun onCreate() {
+        super.onCreate()
+        val (trackingApiUrl, siteId, apiKey) = runBlocking {
+            return@runBlocking try {
+                listOf(
+                    dataStore.data.first()[TRACK_API_URL_KEY],
+                    dataStore.data.first()[SITE_ID],
+                    dataStore.data.first()[API_KEY]
+                )
```

Comment:
We can avoid repetition by using variable, something like follows:

```suggestion
                val data = dataStore.data.first()
                listOf(
                    data[TRACK_API_URL_KEY],
                    data[SITE_ID],
                    data[API_KEY]
                )
```

Reply:
ah ! very good suggestion. It would not just avoid repetition but `data.first()` call is synchronous, so it won't block multiple times either.